File: sql/mysql/Oracle/Antlr4ng/SqlModes.ts
/*
 * Copyright © 2025, Oracle and/or its affiliates
 */

/* eslint-disable no-underscore-dangle */
/* cspell: ignore antlr, longlong, ULONGLONG, MAXDB */

/** SQL modes that control parsing behavior. */

import SqlMode from "./SqlMode.js";

export class SqlModes {
	
    /**
     * Converts a mode string into individual mode flags.
     *
     * @param modes The input string to parse.
     */
    public static sqlModeFromString(modes: string): Set<SqlMode> {
        var result = new Set<SqlMode>();

        const parts = modes.toUpperCase().split(",");
        parts.forEach((mode: string) => {
            if (mode === "ANSI" || mode === "DB2" || mode === "MAXDB" || mode === "MSSQL" || mode === "ORACLE" ||
                mode === "POSTGRESQL") {
                result.add(SqlMode.AnsiQuotes).add(SqlMode.PipesAsConcat).add(SqlMode.IgnoreSpace);
            } else if (mode === "ANSI_QUOTES") {
                result.add(SqlMode.AnsiQuotes);
            } else if (mode === "PIPES_AS_CONCAT") {
                result.add(SqlMode.PipesAsConcat);
            } else if (mode === "NO_BACKSLASH_ESCAPES") {
                result.add(SqlMode.NoBackslashEscapes);
            } else if (mode === "IGNORE_SPACE") {
                result.add(SqlMode.IgnoreSpace);
            } else if (mode === "HIGH_NOT_PRECEDENCE" || mode === "MYSQL323" || mode === "MYSQL40") {
                result.add(SqlMode.HighNotPrecedence);
            }
        });
	return result;
    }
}

export default SqlModes;

File: sql/mysql/Dart/SqlModes.dart
import 'package:antlr4/antlr4.dart';
import 'dart:io';
import 'dart:core';
import 'dart:convert';
import 'dart:collection';
import 'MySQLLexer.dart';
import 'MySQLParser.dart';
import 'SqlMode.dart';

class SqlModes
{
    static HashSet<SqlMode> sqlModeFromString(String modes)
    {
        var result = HashSet<SqlMode>();
        List<String> parts = modes.toUpperCase().split(',');
        for (String mode in parts)
	{
            if (['ANSI', 'DB2', 'MAXDB', 'MSSQL', 'ORACLE', 'POSTGRESQL'].contains(mode))
	    {
		result.add(SqlMode.ansiQuotes);
                result.add(SqlMode.pipesAsConcat);
                result.add(SqlMode.ignoreSpace);
            } else if (mode == 'ANSI_QUOTES') {
                result.add(SqlMode.ansiQuotes);
            } else if (mode == 'PIPES_AS_CONCAT') {
                result.add(SqlMode.pipesAsConcat);
            } else if (mode == 'NO_BACKSLASH_ESCAPES') {
                result.add(SqlMode.noBackslashEscapes);
            } else if (mode == 'IGNORE_SPACE') {
                result.add(SqlMode.ignoreSpace);
            } else if (['HIGH_NOT_PRECEDENCE', 'MYSQL323', 'MYSQL40']
                .contains(mode)) {
                result.add(SqlMode.highNotPrecedence);
            }
        }
        return result;
    }
}
